hydra-node/src/Hydra/Chain/Direct/Wallet.hs

```diff
@@ -365,16 +361,29 @@ coverFee_ pparams systemStart epochInfo lookupUTxO walletUTxO partialTx@Babbage.
             (floor (maxMem * approxMem % totalMem))
             (floor (maxCpu * approxCpu % totalCpu))
 
-findFuelUTxO :: Map TxIn TxOut -> Maybe (TxIn, TxOut)
-findFuelUTxO utxo =
-  let utxosWithDatum = Map.toList $ Map.filter hasMarkerDatum utxo
-      sortingCriteria (_, Babbage.BabbageTxOut _ v _ _) = fst' (gettriples' v)
-      sortedByValue = sortOn sortingCriteria utxosWithDatum
-   in case sortedByValue of
+findFuelOrLargestUTxO :: Map TxIn TxOut -> Maybe (TxIn, TxOut)
+findFuelOrLargestUTxO utxo =
+  case findFuelUTxO utxo of
+    Nothing ->
+      case UTxO.pairs $ UTxO.maxLovelaceUTxO apiUtxo of
         [] -> Nothing
-        as -> Just (List.last as)
+        as ->
+          -- NOTE: pick the **FIRST** value from the sorted list.
+          Just $ bimap toLedgerTxIn toLedgerTxOut (List.head as)
+    Just fuelUTxO -> Just fuelUTxO
  where
-  hasMarkerDatum :: TxOut -> Bool
+  apiUtxo = UTxO.fromPairs $ bimap fromLedgerTxIn fromLedgerTxOut <$> Map.toList utxo
+  findFuelUTxO utxo' =
+    let utxosWithDatum = Map.toList $ Map.filter hasMarkerDatum utxo'
+        sortingCriteria (_, Babbage.BabbageTxOut _ v _ _) = fst' (gettriples' v)
+        sortedByValue = sortOn sortingCriteria utxosWithDatum
+     in case sortedByValue of
+          [] -> Nothing
+          as ->
+            -- NOTE: here we are picking the **LAST** entry we found for no
+            -- particular reason.
```

Comment:
It's not "no particular reason", but it's contrived. Seems like `gettriples'` gives us the lovelace as integer on the fst position and hence this is using the ledger api to sort by lovelace. So the last would be the biggest marked utxo here.